File: libraries/core/android/src/main/java/org/md2k/core/cerebralcortex/cerebralcortexwebapi/models/stream/StreamMetadata.java
package org.md2k.core.cerebralcortex.cerebralcortexwebapi.models.stream;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.HashMap;
import java.util.List;

public class StreamMetadata {
    @SerializedName("name")
    @Expose
    private String name;

    @SerializedName("description")
    @Expose
    private String description;

    @SerializedName("data_descriptor")
    @Expose
    private List<DataDescriptor> dataDescriptors;

    @SerializedName("modules")
    @Expose
    private List<Module> modules;

    public StreamMetadata() {
    }

    public StreamMetadata(String name, String description,
                          List<DataDescriptor> dataDescriptors,
                          List<Module> modules) {
        super();
        this.name = name;
        this.description = description;
        this.dataDescriptors = dataDescriptors;
        this.modules = modules;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public List<DataDescriptor> getDataDescriptors() {
        return dataDescriptors;
    }

    public void setDataDescriptors(List<DataDescriptor> dataDescriptors) {
        this.dataDescriptors = dataDescriptors;
    }

    public List<Module> getModules() {
        return modules;
    }

    public void setModules(List<Module> modules) {
        this.modules = modules;
    }
}

File: libraries/core/lib/controller/datakit/archive/messagepack/stream_metadata.dart
import 'data_descriptor.dart';
import 'module.dart';

class StreamMetadata {
  String name;
  String description;
  List<DataDescriptor> dataDescriptor;
  List<Module> modules;

  StreamMetadata(String name, String description,
      List<DataDescriptor> dataDescriptors, List<Module> modules) {
    this.name = name;
    this.description = description;
    this.dataDescriptor = dataDescriptors;
    this.modules = modules;
  }

  Map toJson() {
    Map<String, dynamic> m = new Map();
    m["name"] = name;
    m["description"] = description;
    m["data_descriptor"] =
        dataDescriptor; //jsonEncode(data_descriptor.map((e) => e.toJson()).toList());
    m["modules"] =
        modules; //jsonEncode(modules.map((e) => e.toJson()).toList());
    return m;
  }

  factory StreamMetadata.fromJson(Map<String, dynamic> json) {
    return StreamMetadata(
        json["name"],
        json["description"],
        (json["data_descriptor"] as List)
            .map((i) => DataDescriptor.fromJson(i))
            .toList(),
        (json["modules"] as List).map((i) => Module.fromJson(i)).toList());
  }
}
